worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1246,6 +1215,28 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
     }
   }
 
+  def writeDataWithExceptionHandling(
+      fileWriter: FileWriter,
+      body: ByteBuf,
+      shuffleKey: String): Future[Unit] = {
+    Future {
+      try {
+        fileWriter.write(body)
+      } catch {
+        case e: AlreadyClosedException =>
+          fileWriter.decrementPendingWrites()
+          val (mapId, attemptId) = getMapAttempt(body)
+          val endedAttempt =
+            if (shuffleMapperAttempts.containsKey(shuffleKey)) {
+              shuffleMapperAttempts.get(shuffleKey).get(mapId)
+            } else -1
+          logWarning(s"Append data failed for task(shuffle $shuffleKey, map $mapId, attempt" +
```

Comment:
There could be two cases when invoke this method: handleMapPartitionPushData and handlePushMergedData. And the original code annotate which case it is here. Not sure how it is helpful but can we retain which case it is here?

Reply:
How did the original code annotate which case it is? Sorry, not sure if I am missing something

Reply:
oh I was referring to [L671](https://github.com/apache/incubator-celeborn/blob/main/worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala#L671) and [L842](https://github.com/apache/incubator-celeborn/blob/main/worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala#L842)

Reply:
Oh, I see. I think now that we're throwing the exception itself, it will be evident in the stack trace anyways which method it is coming from, so might be unnecessary.

Reply:
makes sense, thanks for clarifying.